polaris.shopify.com/content/patterns/date-picking.ts

```diff
@@ -435,387 +474,501 @@ const pattern: MultiVariantPattern = {
           }
           useEffect(() => {
             if (activeDateRange) {
-              setDate({
-                month: activeDateRange.period.since.getMonth(),
-                year: activeDateRange.period.since.getFullYear(),
-              });
               setInputValues({
                 since: formatDate(activeDateRange.period.since),
                 until: formatDate(activeDateRange.period.until),
               });
+              if (activeDateRange.period.since === activeDateRange.period.until) {
+                return;
+              }
+              const clonedDate = new Date(activeDateRange.period.until);
+              clonedDate.setMonth(clonedDate.getMonth() - 1);
+              if (activeDateRange.period.since.getMonth() === month) {
+                if (
+                  activeDateRange.period.since.getMonth() ===
+                  activeDateRange.period.until.getMonth()
+                ) {
+                  clonedDate.setMonth(clonedDate.getMonth() + 1);
+                }
+              }
```

Comment:
```suggestion
              if (
                activeDateRange.period.since.getMonth() !== month ||
                activeDateRange.period.until.getMonth() !== month
              ) {
                clonedDate.setMonth(clonedDate.getMonth() - 1);
              }
```

I'm fairly certain this logic is equivalent, and easier to understand; _If either month has changed, set the month to be 1 month in the past_.

_However_... I don't understand why we would want to do that? What is it fixing?

Also, what about the case where the months match, but they're in different years?

Reply:
The bug we're trying to stop here is random layout changes when in multi-month view. This used to happen because in a date picker that displayed the months [June, July] any selections that started in July would trigger a layout shift, as the useEffect set the visibleMonth to the month and year of the start of the active date range. 

The answer here I think is actually much simpler than both versions we've outlined. 
Actually what we want is to check the month difference between the present visible month in the calendar, and the end of the date range we've selected. If there is no difference, or a difference less than or equal to 1, don't change the visible month or year. Otherwise change it.